codersdk/provisionerdaemons.go

```diff
@@ -193,14 +194,17 @@ type ServeProvisionerDaemonRequest struct {
 // ServeProvisionerDaemon returns the gRPC service for a provisioner daemon
 // implementation. The context is during dial, not during the lifetime of the
 // client. Client should be closed after use.
-func (c *Client) ServeProvisionerDaemon(ctx context.Context, req ServeProvisionerDaemonRequest) (proto.DRPCProvisionerDaemonClient, error) {
+func (c *Client) ServeProvisionerDaemon(ctx context.Context, v *apiversion.APIVersion, req ServeProvisionerDaemonRequest) (proto.DRPCProvisionerDaemonClient, error) {
```

Comment:
I'm not crazy about this parameter. it makes it easier to do the wrong thing (not pass a version) than the right thing (specific `CurrentVersion`).

Reply:
Could refactor to something like the below?

```
func (c *Client) ServeProvisionerDaemon(ctx, context.Context, req ServeProvisionerDaemonRequest)  {
   serveProvisionerDaemonWithVersion(ctx, currentVersion, req)
}
```

Reply:
I ended up restoring the original method signature and just hitting the endpoint with a plain request, which accomplishes the same goal but in a slightly wordier fashion.